parachain/pallets/control/src/tests.rs

```diff
@@ -506,3 +508,34 @@ fn check_sibling_sovereign_account() {
 		);
 	});
 }
+
+#[test]
+fn charge_fee() {
```

Comment:
All the happy case path tests such as the [create_agent test](https://github.com/Snowfork/snowbridge/pull/975/files#diff-494b86672bfad8881d114f1bb5765ab5ef9def9f532a51aa13b80b13f0b06714R9) should assert that the fee was charged or not charged as opposed to one test for all cases if possible. If not possible maybe we should have separate asserts for each case in this test.

Reply:
https://github.com/Snowfork/snowbridge/pull/975/commits/9037a8ce0bcb74a2b231d07c67f5cdb9a8350098